src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -89,7 +90,8 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             if (metricStreamConfiguration is HistogramConfiguration histogramConfiguration)
             {
                 // Validate histogram bounds
-                if (histogramConfiguration.BucketBounds != null && !IsSortedAndDistinct(histogramConfiguration.BucketBounds))
+                if (histogramConfiguration.BucketBounds != null &&
+                    !Utils.IsSortedAndDistinct(histogramConfiguration.BucketBounds))
```

Comment:
I think we need to handle the following cases:
1. if `Double.NaN` is provided in the `BucketBounds`, throw exception.
2. if `Double.NegativeInfinity` or `Double.PositiveInfinity` exist in the `BucketBounds` (and the array is ordered), skip the value.
3. if the resulting `BucketBounds` has no bounds, use `(-Inf, +Inf)` as the single bucket.